core-plugin/src/com/google/gct/idea/debugger/ProjectRepositoryValidator.java

```diff
@@ -105,13 +106,25 @@ public SyncResult checkSyncStashState() {
         for (Debuggee debuggee : debuggees.getDebuggees()) {
           if (myProcessState.getDebuggeeId() != null && myProcessState.getDebuggeeId().equals(debuggee.getId())) {
             foundDebuggee = true;
+            // TODO: This field is deprecated. Consumers should use
+            // ext_source_contexts if it is not empty.
             List<SourceContext> contexts = debuggee.getSourceContexts();
             if (contexts != null) {
               for (SourceContext sourceContext : contexts) {
                 cloudRepo = sourceContext.getCloudRepo();
                 if (cloudRepo != null) {
+                  repoType = GctBundle.getString("clouddebug.cloud.repository");
                   break;
                 }
+                else if (sourceContext.getGerrit() != null) {
+                  repoType = GctBundle.getString("clouddebug.gerrit");
+                }
+                else if (sourceContext.getCloudWorkspace() != null) {
+                  repoType = GctBundle.getString("clouddebug.workspace");
+                }
+                else if (sourceContext.getGit() != null) {
+                  repoType = GctBundle.getString("clouddebug.nongoogle.git");
```

Comment:
The code's a bit of a mess, especially since we have to parse through a lot of optional protobuf fields. This is going to need some refactoring but we should probably do that anyway.